jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/RuleSetNodeInstance.java

```diff
@@ -65,83 +75,112 @@
+                }
+
+                if (dmnResult.hasErrors()) {
+                    String errors = dmnResult.getMessages(Severity.ERROR).stream()
+                            .map(message -> message.toString())
+                            .collect(Collectors.joining(", "));
+
+                    throw new RuntimeException("DMN result errors:: " + errors);
+                }
+                processOutputs(dmnResult.getContext().getAll());
                 triggerCompleted();
-        	}
-    	}
+            } else {
+
+                // first set rule flow group
+                setRuleFlowGroup(resolveRuleFlowGroup(ruleSetNode.getRuleFlowGroup()));
+
+                //proceed
+                for (Entry<String, Object> entry : inputs.entrySet()) {
+                    String inputKey = getRuleFlowGroup() + "_" + getProcessInstance().getId() + "_" + entry.getKey();
+
+                    factHandles.put(inputKey, kruntime.insert(entry.getValue()));
+                }
+
+                if (actAsWaitState()) {
+                    addRuleSetListener();
+                    ((InternalAgenda) getProcessInstance().getKnowledgeRuntime().getAgenda())
+                            .activateRuleFlowGroup(getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId());
+
+                } else {
+                    ((InternalAgenda) getProcessInstance().getKnowledgeRuntime().getAgenda())
+                            .activateRuleFlowGroup(getRuleFlowGroup(), getProcessInstance().getId(), getUniqueId());
+
+                    ((KieSession) getProcessInstance().getKnowledgeRuntime()).fireAllRules();
+
+                    removeEventListeners();
+                    retractFacts();
+                    triggerCompleted();
+                }
+            }
+        } catch (Exception e) {
+            if(!handleException(e, false)) {
+                throw new WorkflowRuntimeException(this, getProcessInstance(), "Unable to execute Action: " + e.getMessage(), e);
+            }
+        }
+    }
+
+    private boolean handleException(Throwable e, boolean handled) {
+        ExceptionScopeInstance exceptionScopeInstance = getExceptionScopeInstance(e);
+        if(exceptionScopeInstance != null) {
+            exceptionScopeInstance.handleException(e.getClass().getName(), e);
+            handled = true;
+        }
+
+        if(!handled && ExceptionUtils.getRootCause(e) != null) {
+            return handleException(ExceptionUtils.getRootCause(e), handled);
+        }
+        return handled;
+    }
+
+    private ExceptionScopeInstance getExceptionScopeInstance(Throwable e) {
+        String exceptionName = e.getClass().getName();
+        Context exceptionContext = getContextContainer().getDefaultContext(ExceptionScope.EXCEPTION_SCOPE).resolveContext(exceptionName);
+        return exceptionContext == null ? null : (ExceptionScopeInstance) getContextInstance(exceptionContext);
```

Comment:
this method should be responsible for dealing with that: https://github.com/kiegroup/jbpm/blob/master/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java#L420 meaning if there is no structureRef given it should fallback to the default error handler. We need to investigate that as I'd prefer to have the backend processing aligned for all node instance types.